app/assets/javascripts/discourse/app/components/emoji-picker.js

```diff
@@ -273,8 +271,23 @@ export default Component.extend({
 
     if (event.key === "Escape") {
       this.onClose(event);
-      const nearestInput = document.querySelector("textarea");
-      nearestInput.focus();
+
+      const fromChatComposer = event.path.find((e) =>
+        e?.classList?.contains("chat-composer-container")
```

Comment:
You should be able to use `event.target.closest()` here.

Reply:
Hmm, I tried that at first, but it doesn't seem to work...